### PR TITLE
Submit HW_2

### PR DESCRIPTION
**Pull request by BossRandy**
JS-SF-7 Homework due 5/31

### CODE REVIEW

**Comment by BossRandy** [comment]
Submit HW_3

File: Randy-bossrandy/HW_2/3-object-homework/app.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+
+
+// Answer Starts Here
+var some_pet = {
+    name: "Mango",
+    species: "Cat",
+    breed: "short-hair",
+    noise: "meow"
+};
```
Comment (larissam) [inline Randy-bossrandy/HW_2/3-object-homework/app.js]: Looking good! 💯

File: Randy-bossrandy/HW_2/4-homework-99-bottles-of-beer_v2/app.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    var startBottles = 99;
+    var endBottles = 1;
+
+    var i = startBottles;
+    do {
+        printLyric(i, startBottles);
+        i--;
+    } while (i >= 0)
+    
+}
```
Comment (larissam) [inline Randy-bossrandy/HW_2/4-homework-99-bottles-of-beer_v2/app.js]: Good job! Nice use of ` window.onload ` as well. :)

File: Randy-bossrandy/HW_2/4-json-homework/json-hw.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    links.push(element["link"]);
+});
+
+/*
+4. Convert the 'links' array into a JSON string. Do a console.log of that JSON string, copy the results from the terminal and paste them into the input box at http://jsonlint.com/
+Did it create a valid JSON string?
+*/
+
+var jsonlinks = JSON.stringify(links);
+console.log(jsonlinks);
```
Comment (larissam) [inline Randy-bossrandy/HW_2/4-json-homework/json-hw.js]: Perfect! 👍

File: Randy-bossrandy/HW_2/4-json-homework/json-hw.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+/*
+2. Iterate (this means 'use a loop') and do a console.log of each of the item titles
+with the date the photo was taken. The first console.log should return the following:
+
+"Edelweiss Air, Airbus A320-214, HB-IJU 2016-02-23T18:48:14Z"
+
+*/
+
+flickrObj["items"].forEach(function (element, idx) {
```
Comment (larissam) [inline Randy-bossrandy/HW_2/4-json-homework/json-hw.js]: It's a matter of stylistic preference, but I tend to leave off the id argument of the callback function if I'm not using it.

File: Randy-bossrandy/HW_2/5-madlibs/js/madlib-console.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    //console.log(temp);
+
+    var hdr = document.getElementById('xForY');
+    hdr.innerHTML = temp;
+
+    favsadded = false;
+}
+
+save.onclick = function () {
+    if (temp.length > 0 && favsadded === false)
```
Comment (larissam) [inline Randy-bossrandy/HW_2/5-madlibs/js/madlib-console.js]: Nice - good job catching blanks and duplicates! 💯

File: Randy-bossrandy/HW_2/5-madlibs/js/madlib-console.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        favs.push(temp);
+
+    favsadded = true;
+}
+
+showfav.onclick = function () {
```
Comment (larissam) [inline Randy-bossrandy/HW_2/5-madlibs/js/madlib-console.js]: Great job!

File: Randy-bossrandy/HW_3/4-homework-99-bottles-of-beer_v3/app.js [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+  For an extra challenge, improve this code to print "1 bottle" rather than "1 bottles." (Hint: read up on if/else
+  conditional statements at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/if...else). 
+*/
+
+//***Change "Bottles" to singlular.
+function bottleReplace(input) {
+    return input.replace(/bottles/g, 'bottle');
+}
+
+function printLyric(bottlesLeft, startBottle) {
```
Comment (larissam) [inline Randy-bossrandy/HW_3/4-homework-99-bottles-of-beer_v3/app.js]: Nice - looks good!

File: Randy-bossrandy/HW_3/5_jquery_homework/app.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+    });
+}
+
+function addRemoveButton(li) {
+    var $removeButton = $('<button>').html("Remove question");
+    $(li).append($removeButton);
+}
+
+$('#surveyList').on('click', 'li button', function (e) {
```
Comment (larissam) [inline Randy-bossrandy/HW_3/5_jquery_homework/app.js]: Awesome! 👍

File: Randy-bossrandy/HW_2/4-json-homework/json-hw.js (hunk 4 above)
Comment (BossRandy) [inline Randy-bossrandy/HW_2/4-json-homework/json-hw.js]: Good point.

**Pull request merged**